private/syntax-path.rkt

```diff
@@ -20,6 +20,8 @@
   [syntax-path-add (-> syntax-path? exact-nonnegative-integer? syntax-path?)]
   [syntax-path-remove-prefix (-> syntax-path? syntax-path? syntax-path?)]
   [syntax-path-neighbors? (-> syntax-path? syntax-path? boolean?)]
+  [syntax-path->string (-> syntax-path? string?)]
```

Comment:
Make this promise to return an `immutable-string?` (that predicate comes from the `racket/mutability` library).